### PR TITLE
Adding symlink for dotnet-compile-native

### PR DESCRIPTION
**Pull request by blackdwarf**
Add symlink for `dotnet-compile-native`. Also, bring the postinstall script in PKG up to par with the Debian package symlinks. 

/cc @Sridhar-MS @brthor

### CODE REVIEW

File: packaging/osx/scripts/postinstall [hunk 1]
```diff
@@ -12,19 +12,21 @@ INSTALL_DESTINATION=$2
...
 ln -s $INSTALL_DESTINATION/bin/dotnet-pack /usr/local/bin/
 ln -s $INSTALL_DESTINATION/bin/dotnet-publish /usr/local/bin/
 ln -s $INSTALL_DESTINATION/bin/dotnet-repl /usr/local/bin/
+ln -s $INSTALL_DESTINATION/bin/dotnet-repl-csi /usr/local/bin/
 ln -s $INSTALL_DESTINATION/bin/dotnet-restore /usr/local/bin/
-ln -s $INSTALL_DESTINATION/bin/dotnet-resgen /usr/local/bin/
 ln -s $INSTALL_DESTINATION/bin/dotnet-run /usr/local/bin/
-ln -s $INSTALL_DESTINATION/bin/dotnet-test /usr/local/bin/
 ln -s $INSTALL_DESTINATION/bin/dotnet-dnx /usr/local/bin/
+ln -s $INSTALL_DESTINATION/bin/dotnet-test /usr/local/bin/
```
Comment (piotrpMSFT) [inline packaging/osx/scripts/postinstall]: Can we alphabetize these? it's tough to match inventory across the many files in which they're listed without this.

**Comment by blackdwarf** [comment]
Moved the PKG to use the system-wide path to avoid symlinking 100+ stuff.

/cc @Sridhar-MS @brthor @piotrpMSFT @livarcocc

**Comment by brthor** [comment]
:+1:

**Comment by blackdwarf** [comment]
@brthor yes, it does. It fixes all of the problems with the symlink resolution problems.

**Comment by blackdwarf** [comment]
@dotnet-bot test OSX Debug Build please

**Comment by brthor** [comment]
@blackdwarf Looks like you'll need to rebase to resolve these conflicts

**Comment by blackdwarf** [comment]
@brthor my life is hell...but ok, will do. ;)

**Comment by piotrpMSFT** [comment]
@blackdwarf thanks!

**Pull request merged**